projects/cdk/directives/auto-focus/autofocus.options.ts

```diff
@@ -0,0 +1,57 @@
+import {
+    ElementRef,
+    inject,
+    InjectionToken,
+    NgZone,
+    Optional,
+    Renderer2,
+    Self,
+} from '@angular/core';
+import {ANIMATION_FRAME} from '@ng-web-apis/common';
+import {TuiFocusableElementAccessor} from '@taiga-ui/cdk/interfaces';
+import {TUI_FOCUSABLE_ITEM_ACCESSOR, TUI_IS_APPLE} from '@taiga-ui/cdk/tokens';
+import {Observable} from 'rxjs';
+
+import {TuiDefaultAutofocusHandler} from './handlers/default.handler';
+import {TuiIosAutofocusHandler} from './handlers/ios.handler';
+
+export interface TuiAutofocusHandler {
+    setFocus(): void;
+}
+
+export const TUI_AUTOFOCUS_HANDLER = new InjectionToken<TuiAutofocusHandler | null>(
+    'Autofocusing handler',
+    {factory: () => null},
+);
+
+// TODO: 3.0 remove in ivy compilation
+export function tuiAutofocusHandlerFactory(
+    tuiFocusableComponent: TuiFocusableElementAccessor | null,
+    elementRef: ElementRef<HTMLElement>,
+    animationFrame$: Observable<number>,
+    renderer: Renderer2,
+    ngZone: NgZone,
+): TuiAutofocusHandler {
+    return inject(TUI_IS_APPLE)
```

Comment:
I know it's more idiomatic to do it like that, but `useClass` with global `navigator` hack allows us not to care about dependencies of these classes. @nsbarsukov @vladimirpotekhin what do you think?
![image](https://user-images.githubusercontent.com/11832552/169292511-a55c0b65-34ff-4f08-bf49-f6ad08fd4cb5.png)

Reply:
useClass with expressions does not compile to ViewEngine